tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -925,6 +925,47 @@ var _ = Describe("odo devfile push command tests", func() {
 		Context("odo push (without -f) is executed", func() {
 			ensureFilesSyncedTest(commonVar.Project, false)
 		})
+
+		When("node-js application is created and pushed with devfile schema 2.2.0", func() {
+			ensureResource := func(output, cpulimit, cpurequest, memoryrequest string) {
+				It("should push the component successfully and check cpuLimit,cpuRequests,memoryRequests", func() {
+					By("check for cpuLimit", func() {
+						podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+						output = helper.Cmd("kubectl", "get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.limits.cpu}'").ShouldPass().Out()
+						Expect(output).To(ContainSubstring(cpulimit))
+					})
+					By("check for cpuRequests", func() {
+						podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+						output = helper.Cmd("kubectl", "get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.requests.cpu}'").ShouldPass().Out()
+						Expect(output).To(ContainSubstring(cpurequest))
+					})
+					By("check for memoryRequests", func() {
+						podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+						output = helper.Cmd("kubectl", "get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.requests.memory}'").ShouldPass().Out()
+						Expect(output).To(ContainSubstring(memoryrequest))
+					})
+				})
+			}
+
+			var output string
+			BeforeEach(func() {
+				helper.Cmd("odo", "create", "--project", commonVar.Project, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-MR-CL-CR.yaml")).ShouldPass()
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+				output = helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass().Out()
+			})
+
+			ensureResource(output, "1", "200m", "512Mi")
```

Comment:
Can you keep the `It` block here as is, just call this function from inside the `It` block. Current format is confusing.
Something like this - 
```go
It("should push the component successfully and check cpuLimit,cpuRequests,memoryRequests", func() {
ensureResource(output, "1", "200m", "512Mi")
})
```

Same for the next block.

Reply:
on it!